### PR TITLE
Fix some implicit type convert bugs

### PR DESCRIPTION
**Pull request by m-schen**
**What type of PR is this?**

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue 
#3251
#3253 
#3254 
#3267 
#3274
#3277 
#3279 
#3370 
#3550 

**What this PR does / why we need it:**

1. add some missed implicit type convert rule 
2. add default scalar and precision for target type when we implicitly cast param as decimal.2. 
3. add special implicit cast logic for concat_ws
4. format all cast error message
5. aggregate function will not do implicit type conversion now.
6. refresh some expected result of bvt cases.
7. error message '[unexpected type : 0]' is [Any] now.

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/sql/plan2/make.go [hunk 1]
```diff
@@ -205,3 +207,19 @@ func MakePlan2DefaultExpr(expr engine.DefaultExpr) *plan.DefaultExpr {
...
+// set a default value for it.
+func rewriteDecimalTypeIfNecessary(typ *plan.Type) *plan.Type {
+	if typ.Id == plan.Type_DECIMAL128 && typ.Scale == 0 && typ.Width == 0 {
+		typ.Scale = 10
+		typ.Width = 38
+		typ.Size = 16
+	}
+	if typ.Id == plan.Type_DECIMAL64 && typ.Scale == 0 && typ.Width == 0 {
+		typ.Scale = 2
+		typ.Width = 6
```
Comment (fengttt) [inline pkg/sql/plan2/make.go:221]: Decimal64 width is 6?

File: pkg/sql/plan2/make.go [hunk 2]
```diff
@@ -205,3 +207,19 @@ func MakePlan2DefaultExpr(expr engine.DefaultExpr) *plan.DefaultExpr {
...
+		typ.Size = 16
```
Comment (fengttt) [inline pkg/sql/plan2/make.go:222]: Size = 8?

Comment (m-schen) [reply pkg/sql/plan2/make.go:222]: oh, I made a mistake when I copy the code. Sorry. It's 8.

File: pkg/sql/plan2/make.go (hunk 1 above)
Comment (m-schen) [reply pkg/sql/plan2/make.go:221]: yeah, they used width to indicate the precision. I don't know why, but it's right here.

**Comment by m-schen** [comment]
There are 3 error message for CAST

1.   can't cast NULL as <target type>
2.  can't cast <value> from <source type> to <target type>
3.  can't cast column from <souce type> to <target type> because one or more values in that column.

File: test/result/function/func_aggr_avg.result [hunk 3]
```diff
@@ -31,20 +31,15 @@ select avg(h) from t1;
 avg(h)
 14.31400000000000000000000
 select avg(i) from t1;
-avg(i)
-20120312.0000
+Function 'avg' with parameters [DATE] will be implemented in future version.
```
Comment (dengn) [inline test/result/function/func_aggr_avg.result:34]: You shouldn't modify the expected result, if new errors are introduced, you need to add new tags for it.

**Review by dengn — CHANGES_REQUESTED** [review]

**Review by fengttt — APPROVED** [review]

**Pull request merged**